ptrace.c

```diff
@@ -543,47 +543,76 @@ bool sm_searchregions(globals_t *vars, scan_match_type_t match_type, const userv
 
     /* check every memory region */
     while (n) {
-        size_t nread = 0;
-        int dots_remaining = NUM_DOTS;
         size_t bytes_at_next_dot;
         size_t bytes_per_dot;
         double progress_per_dot;
 
         /* load the next region */
         r = n->data;
         bytes_per_dot = r->size / NUM_DOTS;
-        bytes_at_next_dot = bytes_per_dot;
+        bytes_at_next_dot = bytes_per_dot * NUM_DOTS;
         progress_per_dot = (double)bytes_per_dot / total_scan_bytes;
 
+/* The maximum logical size is a comfortable 1MiB (increasing it does not help).
+ * The actual allocation is that plus the rounded size of the maximum possible VLT.
+ * This is needed because the last byte might be scanned as max size VLT,
+ * thus need 2^16 - 2 extra bytes after it */
```

Comment:
`2^16 - 2 bytes` sounds like a calculation and can be misleading. Please use `(2 extra bytes after it)`.

Reply:
That IS a calculation. I'll write it as `(2^16 - 2)` for clarity.